### DISCUSSION TITLE
SAN certificate instead of wildcard

### DISCUSSION THREAD

**Discussion by imiddleton1** [amidaware/tacticalrmm#1588] [General]
I did some searching through the documentation and the discussions here but didn't see this mentioned.

Is there a reason that a LE wildcard certificate needs to be used?  If a single LE SAN certificate was used with the 3 DNS names the renewals could be automated without having to do the constant DNS TXT updates every 3 months since LE can authenticate SAN certificates using HTTP instead of DNS.

**Comment by wh1te909** [comment]
Instructions for SANs are here https://docs.tacticalrmm.com/functions/settings_override/#using-your-own-wildcard-ssl-cert

**Reply by imiddleton1** [reply]
I did see that about using your own certificate.  My question was why the default TRMM installation uses a LE wildcard certificate which requires manual updating instead of a LE SAN certificate for which the renewal can be automated with an HTTP-01 challenge.

**Reply by wh1te909** [reply]
wildcard can be automated too using your DNS provider's API. the TXT record method that the installer uses does not require you to expose your instance to the internet, which a lot of people prefer. with HTTP-01 you need to expose port 80

**Reply by imiddleton1** [reply]
Unfortunately, for reasons I can't discuss, we do not use any of the DNS providers with plugins and we cannot change DNS providers.  

Thanks for the info.  I wasn't trying to be critical, just wanted to know if there was a technical reason that a LE wildcard had to be used.  I understand the security implications of exposing port 80 and if that's the reasoning, that's fine.

I'll likely have to provide my own certificate to make this work for me due to my DNS provider constraints.  Either that or do my own LE SAN certificate with an HTTP-01 challenge.  I do realize that this is totally unsupported.

**Reply by dinger1986** [reply]
this is supported https://docs.tacticalrmm.com/functions/settings_override/#using-your-own-wildcard-ssl-cert

**Reply by wh1te909** [reply]
cool, nope no technical reason. personally I use SANs using the instructions in the docs I linked, it functions exactly the same as a wildcard for trmm's purpose. and using your own cert will not make it unsupported, it's just a cert lol. how you get it doesn't matter as long as it's a valid cert.

**Reply by imiddleton1** [reply]
Awesome.  Thanks for the info.  I'm going to try to roll my own cron job to renew a LE SAN with HTTP-01 and if that doesn't work I'll just buy one if necessary.

**Comment by imiddleton1** [comment]
🤦‍♂️  I've been looking through the docs for a week and just now found this: https://docs.tacticalrmm.com/unsupported_scripts/#use-certbot-to-do-acme-challenge-over-http

**Reply by dinger1986** [reply]
Don't just run the script. Use SANs scripts

**Reply by imiddleton1** [reply]
Is there a problem with that script?  or is it just that it uses 3 separate certs?  I really don't care either way.  It's just that I can't use any of the plugins to do DNS auth.

**Reply by wh1te909** [reply]
the nginx configs in that script are outdated and need to be updated to match the current install, so running that script will cause your installation to break. so maybe just use that script as a guideline and yea you can just change the commands to use SANs instead of 3 separate certs

**Reply by imiddleton1** [reply]
Gotcha.  Thanks for the heads up.
